python/ray/train/base_trainer.py

```diff
@@ -46,7 +46,21 @@
 class TrainingFailedError(RuntimeError):
     """An error indicating that training has failed."""
 
-    pass
+    _RESTORE_MSG = (
+        "The Ray Train run failed. Please inspect the previous error messages for a "
+        "cause. After fixing the issue (assuming that the error is not caused by "
+        "your own application logic, but rather an error such as OOM), you can restart "
+        "the run from scratch or continue this run.\n"
+        "To continue this run, you can use: "
+        '`trainer = {trainer_cls_name}.restore("{path}")`.'
+    )
+
+    _FAILURE_CONFIG_MSG = (
+        "To start a new run that will retry on training failures, set "
+        "`air.RunConfig(failure_config=air.FailureConfig(max_failures))` "
+        "in the Trainer's `run_config` with `max_failures > 0`, or `max_failures = 1` "
```

Comment:
Does `max_failures> 0` cover `max_failures=1`?

Reply:
Oh oops, this should be `max_failures=-1` for infinite retry